pandas/tests/test_strings.py

```diff
@@ -838,8 +838,12 @@ def test_contains_for_object_category(self):
         expected = Series([True, False, False, True, False])
         tm.assert_series_equal(result, expected)
 
-    def test_startswith(self):
-        values = Series(["om", np.nan, "foo_nom", "nom", "bar_foo", np.nan, "foo"])
+    # add category dtype parametrizations for GH-36241
+    @pytest.mark.parametrize("dtype", [None, "category"])
+    def test_startswith(self, dtype):
+        values = Series(
+            ["om", np.nan, "foo_nom", "nom", "bar_foo", np.nan, "foo"], dtype=dtype
```

Comment:
also pls tests with pd.NA as well (you can parameterize over the null value as well